src/main/java/frc/robot/subsystems/Climber_Subsystem.java

```diff
@@ -27,29 +28,100 @@
+        // soft limits are a range. Reverse is the low number and forward is the high number
+        int horizontalReverseLimit = 100;
+        int horizontalForwardLimit = 12000;
+        int verticalReverseLimit = 1000;
+        int verticalForwardLimit = 21000;
+        
+        
+        SmartDashboard.putNumber("Horizontal Forward Limit", horizontalForwardLimit);
+        SmartDashboard.putNumber("Horizontal Reverse Limit", horizontalReverseLimit);
+        SmartDashboard.putNumber("Vertical Forward Limit", verticalForwardLimit);
+        SmartDashboard.putNumber("Vertical Reverse Limit", verticalReverseLimit);
+
+        verticalR.configForwardSoftLimitThreshold(verticalForwardLimit);
+        verticalR.configReverseSoftLimitThreshold(verticalReverseLimit);
+        verticalL.configForwardSoftLimitThreshold(verticalForwardLimit);
+        verticalL.configReverseSoftLimitThreshold(verticalReverseLimit);
+
+        horizontalR.configForwardSoftLimitThreshold(horizontalForwardLimit);
+        horizontalR.configReverseSoftLimitThreshold(horizontalReverseLimit);
+        horizontalL.configForwardSoftLimitThreshold(horizontalForwardLimit);
+        horizontalL.configReverseSoftLimitThreshold(horizontalReverseLimit);
+
+
+        verticalR.configForwardSoftLimitEnable(true);
+        verticalR.configReverseSoftLimitEnable(true);
+        verticalL.configForwardSoftLimitEnable(true);
+        verticalL.configReverseSoftLimitEnable(true);
 
-        verticalR.setSelectedSensorPosition(0, 0, 10);
-        verticalL.setSelectedSensorPosition(0, 0, 10);
-        horizontalR.setSelectedSensorPosition(0, 0, 10);
-        horizontalL.setSelectedSensorPosition(0, 0, 10);
+        horizontalL.configForwardSoftLimitEnable(true);
+        horizontalL.configReverseSoftLimitEnable(true);
+        horizontalR.configForwardSoftLimitEnable(true);
+        horizontalR.configReverseSoftLimitEnable(true);
+    }
 
+    public void updateState()
+    {
+        if(Robot.m_stateFlag.getSelected() == null)
+        {
+            return;
+        }
+        if(updateState == null || updateState != Robot.m_stateFlag.getSelected())
+        {
+            horizontalL.setSensorPhase(Robot.m_hlpChooser.getSelected());
+            horizontalR.setSensorPhase(Robot.m_hrpChooser.getSelected());
+            verticalL.setSensorPhase(Robot.m_vlpChooser.getSelected());
+            verticalR.setSensorPhase(Robot.m_vrpChooser.getSelected());
+    
+            horizontalL.setInverted(Robot.m_hliChooser.getSelected());
+            horizontalR.setInverted(Robot.m_hriChooser.getSelected());
+            verticalL.setInverted(Robot.m_vliChooser.getSelected());
+            verticalR.setInverted(Robot.m_vriChooser.getSelected());
+            updateState = Robot.m_stateFlag.getSelected();
+        }
     }
 
     public void verticalClimb(double y_axis) {
         vError = ahrs.getPitch();
 
-        vResponce = (vError * vp);
+        if(y_axis < 0)
```

Comment:
This is unrelated to the state changes. Just something I think we should consider adding. We don't need PID on the vertical climber when we're retracting the legs because the gyro doesn't help us when we're sitting on the platform. Setting the response to 0 will make it just move both legs according to the raw input.